app/controllers/assignments_controller.rb

```diff
@@ -129,6 +132,89 @@ def student_interface
     end
   end
 
+  def peer_review
+    assignment = Assignment.find(params[:id])
+    @assignment = assignment.is_peer_review? ? assignment : assignment.pr_assignment
+    if @assignment.is_hidden
+      render 'shared/http_status',
+             formats: [:html],
+             locals: {
+                 code: '404',
+                 message: HttpStatusHelper::ERROR_CODE['message']['404']
+             },
+             status: 404,
+             layout: false
+      return
+    end
+
+    @student = current_user
+    @grouping = @student.accepted_grouping_for(@assignment.id)
+    @penalty = SubmissionRule.find_by_assignment_id(@assignment.id)
+    @enum_penalty = Period.where(submission_rule_id: @penalty.id).sort
+
+    if @student.section &&
+        !@student.section.section_due_date_for(@assignment.id).nil?
+      @due_date =
+          @student.section.section_due_date_for(@assignment.id).due_date
+    end
+    if @due_date.nil?
+      @due_date = @assignment.due_date
+    end
+    if @grouping.nil?
+      if @assignment.group_max == 1
+        begin
+          # fix for issue #627
+          # currently create_group_for_working_alone_student only returns false
+          # when saving a grouping throws an exception
+          unless @student.create_group_for_working_alone_student(@assignment.id)
+            # if create_group_for_working_alone_student returned false then the student
+            # must have an ( empty ) existing grouping that he is not a member of.
+            # we must delete this grouping for the transaction to succeed.
+            Grouping.find_by_group_id_and_assignment_id( Group.find_by_group_name(@student.user_name), @assignment.id).destroy
```

Comment:
Don't use the `find_by_*` helpers. Use `find_by(group: ..., assignment_id: ...)`